UltraStar Play/Packages/playshared/Runtime/Util/ObjectUtils.cs

```diff
@@ -8,4 +11,15 @@ public static void Swap<T>(ref T a, ref T b)
         a = b;
         b = tmp;
     }
-}
\ No newline at end of file
+
+    public static void ShuffleList<T>(IList<T> list)
+    {
+        int n = list.Count;
+        while (n > 1)
+        {
+            n--;
+            int k = Random.Range(0, n + 1);
+            (list[k], list[n]) = (list[n], list[k]);
```

Comment:
- Uhh, did not know C# can do tuples this way. Crazy nice!
    - Actually, is it a tuple or multiple statments (swapping in array) combined to be atomic? Is it atomic at all?
- Are you sure this mixes the list randomly? To me it feels like items with lower index are swapped more often, possibly more likely to have a new value.